File: java/BubbleSort.java
import java.util.Arrays;
import java.util.Random;

/**
 * Vygeneruje pole náhodných čísel fixnej dĺžky a usporiada ich od najmenšieho po najväčšie.
 */
public class BubbleSort {

    public static void main(String[] args) {
        int numbers[] = new int[6];

        // Najprv potrebujeme vygenerovať náhodné čísla, ktoré budeme chcieť usporiadať
        for (int i = 0; i < numbers.length; i++) {
            // Index v poli začína od nuly, takže pole veľkosti 6 má hodnoty na indexoch 0-5
            numbers[i] = new Random().nextInt(100);
        }
        System.out.println("Numbers to sort: " + Arrays.toString(numbers));

        // Usporiadame jednotlivé čísla prehadzovaním susediacich čísel tak, aby menšie išlo na nižší index v poli
        for (int i = 0; i < numbers.length - 1; i++) {
            for (int j = 0; j < numbers.length - 1 - i; j++) {
                if (numbers[j] > numbers[j + 1]) {
                    int tempNumber = numbers[j + 1];
                    numbers[j + 1] = numbers[j];
                    numbers[j] = tempNumber;
                }
            }
        }
        System.out.println("Sorted numbers: " + Arrays.toString(numbers));
    }
}

File: python/bubble_sort.py
#
# Zoradenie náhodných celých čísel pomocou algoritmu Bubble sort.
# https://en.wikipedia.org/wiki/Bubble_sort
#

import random

numbers = []

for i in range(6):
    numbers.append(random.randint(0, 99))

print('Generated numbers:', numbers)

for i in range(len(numbers) - 1):
    for j in range(len(numbers) - 1 - i):
        if numbers[j] > numbers[j + 1]:
            numbers[j], numbers[j + 1] = numbers[j + 1], numbers[j]
            # Alternative:
            # numbers[j:j+2] = [numbers[j + 1], numbers[j]]
            

print('Ordered numbers:  ', numbers)
